packages/clay-dropdown/src/ClayDropdownBase.soy

```diff
@@ -0,0 +1,355 @@
+{namespace ClayDropdownBase}
+
+/**
+ * This renders a dropdown.
+ */
+{template .render}
+	{@param items: list<?>}
+	{@param triggerLabel: any}
+	{@param? button: ?}
+	{@param? caption: string}
+	{@param? expanded: bool}
+	{@param? helpText: string}
+	{@param? indicatorsPosition: string}
+	{@param? searchable: bool}
+	{@param? spritemap: string}
+	{@param? triggerStyle: string}
+
+	{let $wrapperClasses kind="text"}
+		dropdown
+		{if $expanded}
+			{sp}show
+		{/if}
+	{/let}
+
+	<div class="{$wrapperClasses}">
+		{call .trigger}
+			{param expanded: $expanded /}
+			{param triggerLabel: $triggerLabel /}
+			{param triggerStyle: $triggerStyle /}
+		{/call}
+
+		{call .sections}
+			{param button: $button /}
+			{param caption: $caption /}
+			{param expanded: $expanded /}
+			{param helpText: $helpText /}
+			{param indicatorsPosition: $indicatorsPosition /}
+			{param items: $items /}
+			{param searchable: $searchable /}
+			{param spritemap: $spritemap /}
+		{/call}
+	</div>
+{/template}
+
+/**
+ * Renders a list of items
+ */
+{template .items}
+	{@param items: list<?>}
+	{@param? indicatorsPosition: string}
+	{@param? spritemap: string}
+
+	{foreach $item in $items}
+		{if $item.type == 'group' and $item.items}
+			{call .items}
+				{param indicatorsPosition: $indicatorsPosition /}
+				{param items: $item.items /}
+				{param spritemap: $spritemap /}
+			{/call}
+
+			{if not isLast($item)}
+				{delcall ClayDropdown.item variant="'separator'"}
+					{param label: '' /}
+				{/delcall}
+			{/if}
+		{else}
+			{delcall ClayDropdown.item variant="$item.type"}
+				{param active: $item.active /}
+				{param checked: $item.checked /}
+				{param disabled: $item.disabled /}
+				{param indicatorSymbol: $item.indicatorSymbol /}
+				{param indicatorsPosition: $indicatorsPosition /}
+				{param inputName: $item.inputName /}
+				{param inputValue: $item.inputValue /}
+				{param label: $item.label ?: '' /}
+				{param spritemap: $spritemap /}
+				{param url: $item.url /}
+			{/delcall}
+		{/if}
+	{/foreach}
+{/template}
+
+/**
+ * Renders a search input
+ */
+{template .search}
+	{@param spritemap: string}
+
+	<div class="dropdown-section">
+		<div class="input-group input-group-inset">
+			<div class="input-group-input">
+				<input class="form-control" data-oninput="_handleSearch" placeholder="Search for..." type="text" ref="searchInput">
```

Comment:
Would not it be better to pass the `placeholder` via API? For cases of other languages.

Reply:
We still need to think about how to manage languages like in ClayBadge. I've just opened an issue for that https://github.com/metal/metal-clay-components/issues/95